File: Solutions/_238.cs
public class Solution
{
    public int[] ProductExceptSelf(int[] nums)
    {

        var result = Enumerable.Repeat<int>(1, nums.Length).ToArray();
        var prefix = 1;
        var postfix = 1;

        for (int i = 0; i < nums.Length; i++)
        {
            result[i] = prefix;
            prefix *= nums[i];
        }
        for (int i = nums.Length - 1; i >= 0; i--)
        {
            result[i] *= postfix;
            postfix *= nums[i];
        }

        return result;
    }
}

File: Solutions/_238.go
func productExceptSelf(nums []int) []int {
	result := make([]int, len(nums))
	prefix := 1
	posfix := 1
	for i := 0; i < len(nums); i++ {
		result[i] = prefix
		prefix *= nums[i]
	}
	for i := len(nums) - 1; i >= 0; i-- {
		result[i] *= posfix
		posfix *= nums[i]
	}
	return result
}
